independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -681,31 +708,37 @@ static ScopeInfo getValidScope(Collection<ScopeInfo> stereotypeScopes, Annotatio
             }
         }
 
-        for (MethodInfo observerMethod : syncObserverMethods) {
-            BeanInfo declaringBean = beanClassToBean.get(observerMethod.declaringClass());
-            if (declaringBean != null) {
-                Injection injection = Injection.forObserver(observerMethod, declaringBean.getImplClazz(), this,
-                        injectionPointTransformer);
-                observers.add(new ObserverInfo(declaringBean, observerMethod, injection, false));
-                injectionPoints.addAll(injection.injectionPoints);
+        syncObserverMethods.forEach(
```

Comment:
Pls don't use lambdas if not necessary. A simple `for` cycle is much more readable.

Reply:
thought it's slightly more readable this way than iterating through `Map.Entry<...>` from `syncObserverMethods.entrySet` but will change it if you prefer it this way.